### PR TITLE
[Core] Adding merge to data value container

### PR DESCRIPTION
**Pull request by ddiezrod**
The idea is to use this method to "complete" a property from other property.

### CODE REVIEW

**Review by RiccardoRossi — DISMISSED** [review]
this definitely needs to be discussed. 

what is the expected behaviour?

**Comment by ddiezrod** [comment]
@RiccardoRossi The idea is that I want to have a general property lets say Property 0. Then, I have a particular Property 1 that only has some specific values. I want all the property values from Property 0 to be copied into Property 1 (except if this variable already exists, then it keeps its old values.)

I tried to do this without doing any change in the core, but after discussing it with @pooyan-dadvand I did not find any other way to do it.

**Comment by maceligueta** [comment]
I think that the decison of 'respecting the old value' or 'overwriting the old value' could be left to the end developer, in the form of a true/false flag or coding two different methods.

**Comment by RiccardoRossi** [comment]
my point however is: does this need to be a class method?

**Comment by ddiezrod** [comment]
@RiccardoRossi so.. what is your suggestion? mData is private so I do not know what else you could do...

**Comment by pooyan-dadvand** [comment]
I would go for a class method as it only changes this class members with almost full cohesion.

**Comment by RiccardoRossi** [comment]
ok, about the method. I'll dismiss my review since it is no longer relevant
However please do take into account the comment of @maceligueta

**Comment by philbucher** [comment]
@ddiezrod please add a test before merging this

**Comment by ddiezrod** [comment]
Ok, I will use a local flag as @RiccardoRossi  and @maceligueta proposed. I will have to add then the corresponding .cpp file and the test for this method.

**Comment by ddiezrod** [comment]
test and flag added. Please review @RiccardoRossi @pooyan-dadvand @jrubiogonzalez

File: kratos/containers/data_value_container.cpp
Comment (RiccardoRossi) [inline kratos/containers/data_value_container.cpp]: i would put the if(overwrite_values) outisde of the loop so that it is done once instead at each iteration

Comment (ddiezrod) [reply kratos/containers/data_value_container.cpp]: You mean something like:
```
if overwrite_values:
    for i in rOther.mData
        for j in mData
            overwrite if variable already exist
        push if variable does not exist

else
    for i in rOther.mData
        for j in mData
    push if variable does not exist
```

Comment (ddiezrod) [reply kratos/containers/data_value_container.cpp]: ?

**Comment by ddiezrod** [comment]
@RiccardoRossi Can you approve now?

**Comment by ddiezrod** [comment]
Anyone? @RiccardoRossi @pooyan-dadvand ?

File: kratos/containers/data_value_container.cpp
Comment (RiccardoRossi) [inline kratos/containers/data_value_container.cpp]: why are u using the key here? i think you can use directly 
  
        i->first == j->first

File: kratos/tests/containers/test_data_value_container.cpp
Comment (RiccardoRossi) [inline kratos/tests/containers/test_data_value_container.cpp:62]: just as a comment, but i think you can write

      container_target.Merge(container_origin, DataValueContainer::OVERWRITE_OLD_VALUES)

**Review by RiccardoRossi — DISMISSED** [review]

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**
